iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/GeneratorXml.java

```diff
@@ -71,22 +87,23 @@ protected Generator readRootElementAttributes(GeneratorAdder adder, NetworkXmlRe
         double targetP = XmlUtil.readDoubleAttribute(context.getReader(), "targetP");
         double targetV = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "targetV");
         double targetQ = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "targetQ");
+        if (Double.isNaN(targetQ)) {
+            targetQ = 0.0;
+        }
```

Comment:
I really don't agree with this: he serializer should not show different data from the Java object! Either it is relevant to put `targetQ = 0.0` in the implementation (and I'm not sure it should, it is really different to have `NaN` and `0.0`) or it should not write `targetQ` when it is `NaN` but changing attributes values on the fly during deserialization and serialization is not a good practice.

Furthermore, I don't understand why you changed the structure of reading generator? It seems to be for backwards compatibility but this doesn't seem necessary to me?

Reply:
Can you explain that to me again?

Reply:
Finally, I preserved the structure of reading generator and I avoided ` targetQ = 0.0`.
In generators `targetQ `must be valid when the voltage regulating control is disabled. In this case the generator is considered as an injection and P and Q must be defined and valid.

When a generator is read from an xml file we have the following sequence:

- Read main attributes
- Create the generator and finally
- Set the `regulatingTerminal`.

Initially, the only option was to create the generator with the voltage regulating control disabled. Then targetQ must be set to 0.0 (a temporary valid targetQ) when it is not valid (NaN).

After supporting `useLocalRegulation` a new alternative appears. The regulating control can be created `enabled` by setting the regulatingTerminal to local (RegulatingTerminal is the only attribute read after creating the control). 
That allows to preserve the structure of reading generator and discard `targetQ = 0.0`

`UseLocalRegulation `is only used when the regulating control is enabled. At the third step of the sequence, the local regulatingTerminal is overwritten by the value received in the xml file.